  INFO root [2019-09-12 20:03:25,966] - 
  INFO root [2019-09-12 20:03:25,967] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 20:03:26,181] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 20:03:26,182] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 20:03:26,182] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 20:03:26,321] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 20:03:26,322] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 20:03:26,326] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:58361 )
  INFO root [2019-09-12 20:03:26,326] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:58361, SOMAXCONN=128.
  INFO root [2019-09-12 20:03:26,327] - server-configs:
  INFO root [2019-09-12 20:03:26,327] - 	internalTcpAddr : 192.168.56.1:58361
  INFO root [2019-09-12 20:03:26,327] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 20:03:26,327] - 	componentID : 9000
  WARN root [2019-09-12 20:03:26,598] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-12 20:03:26,687] - Components::findLogger: found logger, addr:192.168.56.1:58359
  INFO root [2019-09-12 20:03:26,747] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:03:26,747] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:03:26,770] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 6620 ----
  INFO root [2019-09-12 20:03:26,771] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 20:03:26,819] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 20:03:26,819] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 20:03:27,255] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 20:03:27,255] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 20:03:27,402] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 20:03:27,412] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 20:03:28,096] - ---- loginapp is running ----
 DEBUG root [2019-09-12 20:03:28,096] - Components::process(): Request for the process of identity...
  WARN root [2019-09-12 20:03:34,215] - Forwarding logs to logger[192.168.56.1:58359/0/0/0]...
 ERROR root [2019-09-12 20:03:40,615] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=213689323109, lastReceivedTime=213658122337, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-12 20:03:40,617] - The following logs sent to logger failed:
  WARN root [2019-09-12 20:03:40,618] - ==> [2019-09-12 20:03:40 613] TCPPacketReceiver::processPendingEvents(192.168.56.1:58359): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:03:40,618] - ==> [2019-09-12 20:03:40 616] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=213689323109, lastReceivedTime=213658122337, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-12 20:04:37,925] - 
  INFO root [2019-09-12 20:04:37,927] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 20:04:37,928] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 20:04:37,928] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 20:04:37,928] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 20:04:37,949] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 20:04:37,949] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 20:04:37,954] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:58384 )
  INFO root [2019-09-12 20:04:37,954] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:58384, SOMAXCONN=128.
  INFO root [2019-09-12 20:04:37,955] - server-configs:
  INFO root [2019-09-12 20:04:37,955] - 	internalTcpAddr : 192.168.56.1:58384
  INFO root [2019-09-12 20:04:37,955] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 20:04:37,955] - 	componentID : 9000
  WARN root [2019-09-12 20:04:38,174] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-12 20:04:38,375] - Components::findLogger: found logger, addr:192.168.56.1:58378
  INFO root [2019-09-12 20:04:38,375] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:04:38,376] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:04:38,381] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 8656 ----
  INFO root [2019-09-12 20:04:38,381] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 20:04:38,382] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 20:04:38,382] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 20:04:38,579] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 20:04:38,580] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 20:04:38,591] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 20:04:38,591] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 20:04:38,983] - ---- loginapp is running ----
 DEBUG root [2019-09-12 20:04:38,983] - Components::process(): Request for the process of identity...
  WARN root [2019-09-12 20:04:45,101] - Forwarding logs to logger[192.168.56.1:58378/0/0/0]...
  INFO root [2019-09-12 20:04:54,159] - 
  INFO root [2019-09-12 20:04:54,178] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 20:04:54,178] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 20:04:54,179] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 20:04:54,179] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 20:04:54,229] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 20:04:54,229] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 20:04:54,236] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:58408 )
  INFO root [2019-09-12 20:04:54,236] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:58408, SOMAXCONN=128.
  INFO root [2019-09-12 20:04:54,236] - server-configs:
  INFO root [2019-09-12 20:04:54,236] - 	internalTcpAddr : 192.168.56.1:58408
  INFO root [2019-09-12 20:04:54,237] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 20:04:54,237] - 	componentID : 9000
  WARN root [2019-09-12 20:04:54,493] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-12 20:04:54,774] - Components::findLogger: found logger, addr:192.168.56.1:58410
  INFO root [2019-09-12 20:04:54,775] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:04:54,775] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:04:54,780] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 3264 ----
  INFO root [2019-09-12 20:04:54,781] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 20:04:54,782] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 20:04:54,782] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 20:04:54,986] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 20:04:54,986] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 20:04:54,997] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 20:04:54,997] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 20:04:55,399] - ---- loginapp is running ----
 DEBUG root [2019-09-12 20:04:55,399] - Components::process(): Request for the process of identity...
  WARN root [2019-09-12 20:05:01,516] - Forwarding logs to logger[192.168.56.1:58410/0/0/0]...
 ERROR root [2019-09-12 20:05:06,692] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=213882996432, lastReceivedTime=213856184104, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-12 20:05:06,696] - The following logs sent to logger failed:
  INFO root [2019-09-12 20:05:06,696] - ==> [2019-09-12 20:05:06 678] Components::findComponents: register self to dbmgr...
  WARN root [2019-09-12 20:05:06,696] - ==> [2019-09-12 20:05:06 689] TCPPacketReceiver::processPendingEvents(192.168.56.1:58410): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:05:06,697] - ==> [2019-09-12 20:05:06 693] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=213882996432, lastReceivedTime=213856184104, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-12 20:05:06,697] - Components::findComponents: register self to baseappmgr...
  INFO root [2019-09-12 20:05:06,705] - Components::process(): Found all the components!
  WARN root [2019-09-12 20:05:06,788] - TCPPacketReceiver::processPendingEvents(192.168.56.1:58405): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:05:06,790] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=213883217303, lastReceivedTime=213882963615, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-12 20:05:06,839] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-12 20:05:06,839] - onLoginAppShutDown()
  INFO root [2019-09-12 20:05:06,844] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-12 20:05:06,844] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-12 20:05:06,854] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-12 20:05:06,854] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-12 20:05:07,155] - ThreadPool::destroy(): waiting for thread(2)[0x1961000,0x19606a8,], try=1
  INFO root [2019-09-12 20:05:08,382] - 
  INFO root [2019-09-12 20:05:08,383] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 20:05:08,383] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 20:05:08,384] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 20:05:08,384] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 20:05:08,400] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 20:05:08,400] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 20:05:08,405] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:58429 )
  INFO root [2019-09-12 20:05:08,406] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:58429, SOMAXCONN=128.
  INFO root [2019-09-12 20:05:08,406] - server-configs:
  INFO root [2019-09-12 20:05:08,406] - 	internalTcpAddr : 192.168.56.1:58429
  INFO root [2019-09-12 20:05:08,406] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 20:05:08,406] - 	componentID : 9000
  WARN root [2019-09-12 20:05:08,642] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-12 20:05:10,153] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 11900 ----
  INFO root [2019-09-12 20:05:10,154] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 20:05:10,154] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 20:05:10,155] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 20:05:10,358] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 20:05:10,358] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 20:05:10,371] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 20:05:10,371] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 20:05:10,783] - ---- loginapp is running ----
 DEBUG root [2019-09-12 20:05:10,783] - Components::process(): Request for the process of identity...
  INFO root [2019-09-12 20:05:17,911] - Components::findComponents: find logger(1)...
  INFO root [2019-09-12 20:05:17,918] - Components::findComponents: found logger, addr:192.168.56.1:58436
  INFO root [2019-09-12 20:05:17,919] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:05:17,919] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:05:17,920] - Components::findComponents: find dbmgr(2)...
  INFO root [2019-09-12 20:05:18,023] - Components::findComponents: found dbmgr, addr:192.168.56.1:58432
  INFO root [2019-09-12 20:05:18,023] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:05:18,023] - Components::addComponent[dbmgr], uid=2082, componentID=4000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:05:18,023] - Components::findComponents: found dbmgr, addr:192.168.56.1:58432
  WARN root [2019-09-12 20:05:19,023] - Forwarding logs to logger[192.168.56.1:58436/0/0/0]...
 ERROR root [2019-09-12 20:05:19,679] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=213912217896, lastReceivedTime=213908258667, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-12 20:05:19,682] - The following logs sent to logger failed:
  WARN root [2019-09-12 20:05:19,683] - ==> [2019-09-12 20:05:19 678] TCPPacketReceiver::processPendingEvents(192.168.56.1:58436): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:05:19,683] - ==> [2019-09-12 20:05:19 680] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=213912217896, lastReceivedTime=213908258667, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-12 20:05:20,031] - Components::findComponents: find baseappmgr(1)...
  INFO root [2019-09-12 20:06:08,643] - 
  INFO root [2019-09-12 20:06:08,643] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 20:06:08,643] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 20:06:08,644] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 20:06:08,644] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 20:06:08,661] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 20:06:08,661] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 20:06:08,669] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:58450 )
  INFO root [2019-09-12 20:06:08,669] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:58450, SOMAXCONN=128.
  INFO root [2019-09-12 20:06:08,670] - server-configs:
  INFO root [2019-09-12 20:06:08,670] - 	internalTcpAddr : 192.168.56.1:58450
  INFO root [2019-09-12 20:06:08,670] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 20:06:08,671] - 	componentID : 9000
  WARN root [2019-09-12 20:06:08,895] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-12 20:06:09,211] - Components::findLogger: found logger, addr:192.168.56.1:58457
  INFO root [2019-09-12 20:06:09,211] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:06:09,211] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:06:09,216] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 13992 ----
  INFO root [2019-09-12 20:06:09,216] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 20:06:09,217] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 20:06:09,217] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 20:06:09,415] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 20:06:09,415] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 20:06:09,426] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 20:06:09,427] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 20:06:09,817] - ---- loginapp is running ----
 DEBUG root [2019-09-12 20:06:09,818] - Components::process(): Request for the process of identity...
  WARN root [2019-09-12 20:06:15,922] - Forwarding logs to logger[192.168.56.1:58457/0/0/0]...
 ERROR root [2019-09-12 20:11:38,710] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=214765037856, lastReceivedTime=214697781937, inactivityExceptionPeriod=134887325)
  INFO root [2019-09-12 20:11:38,713] - The following logs sent to logger failed:
  WARN root [2019-09-12 20:11:38,713] - ==> [2019-09-12 20:11:38 706] TCPPacketReceiver::processPendingEvents(192.168.56.1:58457): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:11:38,713] - ==> [2019-09-12 20:11:38 711] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=214765037856, lastReceivedTime=214697781937, inactivityExceptionPeriod=134887325)
  WARN root [2019-09-12 20:11:38,808] - TCPPacketReceiver::processPendingEvents(192.168.56.1:58454): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:11:38,812] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=214765266446, lastReceivedTime=214765177678, inactivityExceptionPeriod=134887325)
  INFO root [2019-09-12 20:11:38,822] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-12 20:11:38,828] - onLoginAppShutDown()
  INFO root [2019-09-12 20:11:38,828] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-12 20:11:38,828] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-12 20:11:38,846] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-12 20:11:38,846] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-12 20:11:39,147] - ThreadPool::destroy(): waiting for thread(2)[0x19d26f8,0x19d27c8,], try=1
  INFO root [2019-09-12 20:25:23,598] - 
  INFO root [2019-09-12 20:25:23,598] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 20:25:23,599] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 20:25:23,600] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 20:25:23,600] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 20:25:23,617] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 20:25:23,618] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 20:25:23,623] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:58516 )
  INFO root [2019-09-12 20:25:23,623] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:58516, SOMAXCONN=128.
  INFO root [2019-09-12 20:25:23,623] - server-configs:
  INFO root [2019-09-12 20:25:23,623] - 	internalTcpAddr : 192.168.56.1:58516
  INFO root [2019-09-12 20:25:23,624] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 20:25:23,624] - 	componentID : 9000
  WARN root [2019-09-12 20:25:23,842] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-12 20:25:25,346] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 14124 ----
  INFO root [2019-09-12 20:25:25,347] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 20:25:25,347] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 20:25:25,347] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 20:25:25,619] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 20:25:25,619] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 20:25:25,632] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 20:25:25,632] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 20:25:26,059] - ---- loginapp is running ----
 DEBUG root [2019-09-12 20:25:26,060] - Components::process(): Request for the process of identity...
  INFO root [2019-09-12 20:25:33,193] - Components::findComponents: find logger(1)...
  INFO root [2019-09-12 20:25:33,231] - Components::findComponents: found logger, addr:192.168.56.1:58522
  INFO root [2019-09-12 20:25:33,231] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:25:33,231] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:25:33,232] - Components::findComponents: find dbmgr(2)...
  INFO root [2019-09-12 20:25:33,336] - Components::findComponents: found dbmgr, addr:192.168.56.1:58517
  INFO root [2019-09-12 20:25:33,336] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:25:33,336] - Components::addComponent[dbmgr], uid=2082, componentID=4000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:25:33,342] - Components::findComponents: found dbmgr, addr:192.168.56.1:58517
  WARN root [2019-09-12 20:25:34,343] - Forwarding logs to logger[192.168.56.1:58522/0/0/0]...
 ERROR root [2019-09-12 20:28:32,105] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=217045177442, lastReceivedTime=217026490264, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-12 20:28:32,110] - The following logs sent to logger failed:
  WARN root [2019-09-12 20:28:32,111] - ==> [2019-09-12 20:28:32 101] TCPPacketReceiver::processPendingEvents(192.168.56.1:58522): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:28:32,111] - ==> [2019-09-12 20:28:32 106] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=217045177442, lastReceivedTime=217026490264, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-12 20:28:32,207] - TCPPacketReceiver::processPendingEvents(192.168.56.1:58517): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:28:32,211] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=217045415740, lastReceivedTime=217026280725, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-12 20:28:32,223] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-12 20:28:32,228] - onLoginAppShutDown()
  INFO root [2019-09-12 20:28:32,228] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-12 20:28:32,229] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-12 20:28:32,247] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-12 20:28:32,248] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-12 20:28:32,549] - ThreadPool::destroy(): waiting for thread(2)[0x1a6d088,0x1a6d0f0,], try=1
  INFO root [2019-09-12 20:32:25,235] - 
  INFO root [2019-09-12 20:32:25,242] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 20:32:25,245] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 20:32:25,245] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 20:32:25,245] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 20:32:25,288] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 20:32:25,288] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 20:32:25,295] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:58581 )
  INFO root [2019-09-12 20:32:25,295] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:58581, SOMAXCONN=128.
  INFO root [2019-09-12 20:32:25,295] - server-configs:
  INFO root [2019-09-12 20:32:25,295] - 	internalTcpAddr : 192.168.56.1:58581
  INFO root [2019-09-12 20:32:25,295] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 20:32:25,296] - 	componentID : 9000
  WARN root [2019-09-12 20:32:25,505] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-12 20:32:25,797] - Components::findLogger: found logger, addr:192.168.56.1:58574
  INFO root [2019-09-12 20:32:25,797] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:32:25,797] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:32:25,803] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 2704 ----
  INFO root [2019-09-12 20:32:25,804] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 20:32:25,804] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 20:32:25,805] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 20:32:26,000] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 20:32:26,000] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 20:32:26,012] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 20:32:26,012] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 20:32:26,405] - ---- loginapp is running ----
 DEBUG root [2019-09-12 20:32:26,405] - Components::process(): Request for the process of identity...
  WARN root [2019-09-12 20:32:32,520] - Forwarding logs to logger[192.168.56.1:58574/0/0/0]...
 ERROR root [2019-09-12 20:36:14,907] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=218086483173, lastReceivedTime=218041995470, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-12 20:36:14,913] - The following logs sent to logger failed:
  WARN root [2019-09-12 20:36:14,913] - ==> [2019-09-12 20:36:14 903] TCPPacketReceiver::processPendingEvents(192.168.56.1:58574): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:36:14,913] - ==> [2019-09-12 20:36:14 909] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=218086483173, lastReceivedTime=218041995470, inactivityExceptionPeriod=134887628)
  WARN root [2019-09-12 20:36:15,033] - TCPPacketReceiver::processPendingEvents(192.168.56.1:58580): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:36:15,039] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=218086778076, lastReceivedTime=218042341699, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-12 20:36:15,056] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-12 20:36:15,062] - onLoginAppShutDown()
  INFO root [2019-09-12 20:36:15,063] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-12 20:36:15,064] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-12 20:36:15,079] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-12 20:36:15,080] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-12 20:36:15,382] - ThreadPool::destroy(): waiting for thread(2)[0x1ac3aa8,0x1ac3cb0,], try=1
  INFO root [2019-09-12 20:37:07,234] - 
  INFO root [2019-09-12 20:37:07,249] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 20:37:07,250] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 20:37:07,250] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 20:37:07,250] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 20:37:07,343] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 20:37:07,344] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 20:37:07,348] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:58620 )
  INFO root [2019-09-12 20:37:07,349] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:58620, SOMAXCONN=128.
  INFO root [2019-09-12 20:37:07,349] - server-configs:
  INFO root [2019-09-12 20:37:07,349] - 	internalTcpAddr : 192.168.56.1:58620
  INFO root [2019-09-12 20:37:07,349] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 20:37:07,349] - 	componentID : 9000
  WARN root [2019-09-12 20:37:07,573] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  WARN root [2019-09-12 20:37:07,913] - BundleBroadcast::BundleBroadcast: Cannot bind to port 20782, errorno=10048
  INFO root [2019-09-12 20:37:07,916] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 2052 ----
  INFO root [2019-09-12 20:37:07,917] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 20:37:07,917] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 20:37:07,918] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 20:37:08,118] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 20:37:08,119] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 20:37:08,129] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 20:37:08,129] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 20:37:08,516] - ---- loginapp is running ----
 DEBUG root [2019-09-12 20:37:08,517] - Components::process(): Request for the process of identity...
  INFO root [2019-09-12 20:37:15,650] - Components::findComponents: find logger(1)...
  INFO root [2019-09-12 20:37:15,758] - Components::findComponents: found logger, addr:192.168.56.1:58616
  INFO root [2019-09-12 20:37:15,758] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:37:15,758] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:37:15,759] - Components::findComponents: find dbmgr(2)...
  INFO root [2019-09-12 20:37:15,769] - Components::findComponents: found dbmgr, addr:192.168.56.1:58615
  INFO root [2019-09-12 20:37:15,770] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:37:15,770] - Components::addComponent[dbmgr], uid=2082, componentID=4000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:37:15,779] - Components::findComponents: found logger, addr:192.168.56.1:58616
  INFO root [2019-09-12 20:37:15,792] - Components::findComponents: found dbmgr, addr:192.168.56.1:58615
  WARN root [2019-09-12 20:37:16,792] - Forwarding logs to logger[192.168.56.1:58616/0/0/0]...
 ERROR root [2019-09-12 20:39:07,965] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=218475863835, lastReceivedTime=218474362363, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-12 20:39:07,973] - The following logs sent to logger failed:
  WARN root [2019-09-12 20:39:07,973] - ==> [2019-09-12 20:39:07 958] TCPPacketReceiver::processPendingEvents(192.168.56.1:58616): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:39:07,974] - ==> [2019-09-12 20:39:07 967] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=218475863835, lastReceivedTime=218474362363, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-12 20:39:08,157] - TCPPacketReceiver::processPendingEvents(192.168.56.1:58615): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:39:08,173] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=218476330040, lastReceivedTime=218474343443, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-12 20:39:08,194] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-12 20:39:08,204] - onLoginAppShutDown()
  INFO root [2019-09-12 20:39:08,205] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-12 20:39:08,205] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-12 20:39:08,223] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-12 20:39:08,224] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-12 20:39:08,526] - ThreadPool::destroy(): waiting for thread(2)[0x1c3a4b0,0x1c3aa60,], try=1
 DEBUG root [2019-09-12 20:39:08,935] - TPThread::~TPThread(): 0x1c3a4b0
 DEBUG root [2019-09-12 20:39:08,935] - TPThread::~TPThread(): 0x1c3aa60
 DEBUG root [2019-09-12 20:39:08,936] - ThreadPool::destroy(): successfully!
 DEBUG root [2019-09-12 20:39:09,071] - Bundle::destroyObjPool(): size 36.
 DEBUG root [2019-09-12 20:39:09,071] - Channel::destroyObjPool(): size 15.
 DEBUG root [2019-09-12 20:39:09,072] - TCPPacket::destroyObjPool(): size 19.
 DEBUG root [2019-09-12 20:39:09,072] - UDPPacket::destroyObjPool(): size 1.
 DEBUG root [2019-09-12 20:39:09,072] - EndPoint::destroyObjPool(): size 14.
 DEBUG root [2019-09-12 20:39:09,072] - Address::destroyObjPool(): size 0.
 DEBUG root [2019-09-12 20:39:09,073] - TCPPacketReceiver::destroyObjPool(): size 0.
 DEBUG root [2019-09-12 20:39:09,073] - UDPPacketReceiver::destroyObjPool(): size 0.
  INFO root [2019-09-12 20:39:09,073] - loginapp(9000) has shut down.
  INFO root [2019-09-12 20:42:17,117] - 
  INFO root [2019-09-12 20:42:17,120] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 20:42:17,121] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 20:42:17,122] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 20:42:17,122] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 20:42:17,163] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 20:42:17,163] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 20:42:17,167] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:58674 )
  INFO root [2019-09-12 20:42:17,167] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:58674, SOMAXCONN=128.
  INFO root [2019-09-12 20:42:17,167] - server-configs:
  INFO root [2019-09-12 20:42:17,168] - 	internalTcpAddr : 192.168.56.1:58674
  INFO root [2019-09-12 20:42:17,169] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 20:42:17,169] - 	componentID : 9000
  WARN root [2019-09-12 20:42:17,393] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-12 20:42:17,815] - Components::findLogger: found logger, addr:192.168.56.1:58667
  INFO root [2019-09-12 20:42:17,816] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:42:17,816] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:42:17,822] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 14840 ----
  INFO root [2019-09-12 20:42:17,822] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 20:42:17,823] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 20:42:17,823] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 20:42:18,102] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 20:42:18,103] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 20:42:18,114] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 20:42:18,115] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 20:42:18,520] - ---- loginapp is running ----
 DEBUG root [2019-09-12 20:42:18,521] - Components::process(): Request for the process of identity...
  WARN root [2019-09-12 20:42:24,665] - Forwarding logs to logger[192.168.56.1:58667/0/0/0]...
 ERROR root [2019-09-12 20:44:16,957] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=219171095764, lastReceivedTime=219103873758, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-12 20:44:16,964] - The following logs sent to logger failed:
  WARN root [2019-09-12 20:44:16,965] - ==> [2019-09-12 20:44:16 958] TCPPacketReceiver::processPendingEvents(192.168.56.1:58667): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:44:16,965] - ==> [2019-09-12 20:44:16 958] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=219171095764, lastReceivedTime=219103873758, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-12 20:44:17,081] - TCPPacketReceiver::processPendingEvents(192.168.56.1:58669): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:44:17,086] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=219171383277, lastReceivedTime=219103989904, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-12 20:44:17,108] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-12 20:44:17,125] - onLoginAppShutDown()
  INFO root [2019-09-12 20:44:17,127] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-12 20:44:17,129] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-12 20:44:17,172] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-12 20:44:17,173] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-12 20:44:17,475] - ThreadPool::destroy(): waiting for thread(2)[0x1f913f0,0x1f91db0,], try=1
  INFO root [2019-09-12 20:45:59,789] - 
  INFO root [2019-09-12 20:45:59,790] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 20:45:59,791] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 20:45:59,792] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 20:45:59,792] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 20:45:59,823] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 20:45:59,823] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 20:45:59,830] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:58724 )
  INFO root [2019-09-12 20:45:59,831] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:58724, SOMAXCONN=128.
  INFO root [2019-09-12 20:45:59,831] - server-configs:
  INFO root [2019-09-12 20:45:59,831] - 	internalTcpAddr : 192.168.56.1:58724
  INFO root [2019-09-12 20:45:59,831] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 20:45:59,831] - 	componentID : 9000
  WARN root [2019-09-12 20:46:00,060] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  WARN root [2019-09-12 20:46:00,403] - BundleBroadcast::BundleBroadcast: Cannot bind to port 20947, errorno=10048
  INFO root [2019-09-12 20:46:00,407] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 11644 ----
  INFO root [2019-09-12 20:46:00,408] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 20:46:00,408] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 20:46:00,409] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 20:46:00,619] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 20:46:00,619] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 20:46:00,630] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 20:46:00,631] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 20:46:01,051] - ---- loginapp is running ----
 DEBUG root [2019-09-12 20:46:01,051] - Components::process(): Request for the process of identity...
  INFO root [2019-09-12 20:46:08,192] - Components::findComponents: find logger(1)...
  INFO root [2019-09-12 20:46:08,283] - Components::findComponents: found logger, addr:192.168.56.1:58720
  INFO root [2019-09-12 20:46:08,283] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:46:08,283] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:46:08,284] - Components::findComponents: find dbmgr(2)...
  INFO root [2019-09-12 20:46:08,387] - Components::findComponents: found dbmgr, addr:192.168.56.1:58718
  INFO root [2019-09-12 20:46:08,387] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:46:08,388] - Components::addComponent[dbmgr], uid=2082, componentID=4000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:46:08,388] - Components::findComponents: found dbmgr, addr:192.168.56.1:58718
  WARN root [2019-09-12 20:46:09,389] - Forwarding logs to logger[192.168.56.1:58720/0/0/0]...
 ERROR root [2019-09-12 20:48:37,524] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=219757369512, lastReceivedTime=219739942616, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-12 20:48:37,526] - The following logs sent to logger failed:
  WARN root [2019-09-12 20:48:37,526] - ==> [2019-09-12 20:48:37 523] TCPPacketReceiver::processPendingEvents(192.168.56.1:58720): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:48:37,527] - ==> [2019-09-12 20:48:37 525] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=219757369512, lastReceivedTime=219739942616, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-12 20:48:37,638] - TCPPacketReceiver::processPendingEvents(192.168.56.1:58718): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:48:37,641] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=219757632056, lastReceivedTime=219739830114, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-12 20:48:37,647] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-12 20:48:37,653] - onLoginAppShutDown()
  INFO root [2019-09-12 20:48:37,654] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-12 20:48:37,654] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-12 20:48:37,669] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-12 20:48:37,669] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-12 20:48:37,969] - ThreadPool::destroy(): waiting for thread(2)[0x18cdb48,0x18cd1f0,], try=1
  INFO root [2019-09-12 20:51:14,358] - 
  INFO root [2019-09-12 20:51:14,360] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 20:51:14,361] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 20:51:14,361] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 20:51:14,363] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 20:51:14,412] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 20:51:14,412] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 20:51:14,416] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:58764 )
  INFO root [2019-09-12 20:51:14,416] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:58764, SOMAXCONN=128.
  INFO root [2019-09-12 20:51:14,416] - server-configs:
  INFO root [2019-09-12 20:51:14,416] - 	internalTcpAddr : 192.168.56.1:58764
  INFO root [2019-09-12 20:51:14,421] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 20:51:14,421] - 	componentID : 9000
  WARN root [2019-09-12 20:51:14,648] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  WARN root [2019-09-12 20:51:15,095] - BundleBroadcast::BundleBroadcast: Cannot bind to port 20470, errorno=10048
  INFO root [2019-09-12 20:51:15,099] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 10776 ----
  INFO root [2019-09-12 20:51:15,099] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 20:51:15,099] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 20:51:15,100] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 20:51:15,316] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 20:51:15,316] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 20:51:15,327] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 20:51:15,328] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 20:51:15,753] - ---- loginapp is running ----
 DEBUG root [2019-09-12 20:51:15,753] - Components::process(): Request for the process of identity...
  INFO root [2019-09-12 20:51:22,894] - Components::findComponents: find logger(1)...
  INFO root [2019-09-12 20:51:22,910] - Components::findComponents: found logger, addr:192.168.56.1:58759
  INFO root [2019-09-12 20:51:22,910] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:51:22,910] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:51:22,912] - Components::findComponents: find dbmgr(2)...
  INFO root [2019-09-12 20:51:22,914] - Components::findComponents: found logger, addr:192.168.56.1:58759
  INFO root [2019-09-12 20:51:22,920] - Components::findComponents: found dbmgr, addr:192.168.56.1:58762
  INFO root [2019-09-12 20:51:22,920] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:51:22,920] - Components::addComponent[dbmgr], uid=2082, componentID=4000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:51:23,023] - Components::findComponents: found dbmgr, addr:192.168.56.1:58762
  WARN root [2019-09-12 20:51:24,024] - Forwarding logs to logger[192.168.56.1:58759/0/0/0]...
 ERROR root [2019-09-12 20:54:08,335] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=220501694334, lastReceivedTime=220447713710, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-12 20:54:08,340] - The following logs sent to logger failed:
  WARN root [2019-09-12 20:54:08,340] - ==> [2019-09-12 20:54:08 334] TCPPacketReceiver::processPendingEvents(192.168.56.1:58759): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:54:08,340] - ==> [2019-09-12 20:54:08 335] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=220501694334, lastReceivedTime=220447713710, inactivityExceptionPeriod=134887628)
  WARN root [2019-09-12 20:54:08,438] - TCPPacketReceiver::processPendingEvents(192.168.56.1:58762): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:54:08,440] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=220501933129, lastReceivedTime=220447896950, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-12 20:54:08,467] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-12 20:54:08,468] - onLoginAppShutDown()
  INFO root [2019-09-12 20:54:08,468] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-12 20:54:08,468] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-12 20:54:08,488] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-12 20:54:08,488] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-12 20:54:08,789] - ThreadPool::destroy(): waiting for thread(2)[0x1d2d0f8,0x1d2d710,], try=1
  INFO root [2019-09-12 20:54:10,199] - 
  INFO root [2019-09-12 20:54:10,201] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 20:54:10,203] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 20:54:10,203] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 20:54:10,203] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 20:54:10,228] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 20:54:10,228] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 20:54:10,232] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:58817 )
  INFO root [2019-09-12 20:54:10,232] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:58817, SOMAXCONN=128.
  INFO root [2019-09-12 20:54:10,232] - server-configs:
  INFO root [2019-09-12 20:54:10,232] - 	internalTcpAddr : 192.168.56.1:58817
  INFO root [2019-09-12 20:54:10,232] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 20:54:10,232] - 	componentID : 9000
  WARN root [2019-09-12 20:54:10,498] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-12 20:54:10,644] - Components::findLogger: found logger, addr:192.168.56.1:58812
  INFO root [2019-09-12 20:54:10,644] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:54:10,644] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:54:10,651] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 11596 ----
  INFO root [2019-09-12 20:54:10,652] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 20:54:10,653] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 20:54:10,653] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 20:54:10,862] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 20:54:10,863] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 20:54:10,874] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 20:54:10,875] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 20:54:11,382] - ---- loginapp is running ----
 DEBUG root [2019-09-12 20:54:11,383] - Components::process(): Request for the process of identity...
  WARN root [2019-09-12 20:54:17,523] - Forwarding logs to logger[192.168.56.1:58812/0/0/0]...
 ERROR root [2019-09-12 20:55:55,106] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=220741930162, lastReceivedTime=220708170330, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-12 20:55:55,110] - The following logs sent to logger failed:
  WARN root [2019-09-12 20:55:55,110] - ==> [2019-09-12 20:55:55 104] TCPPacketReceiver::processPendingEvents(192.168.56.1:58812): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:55:55,111] - ==> [2019-09-12 20:55:55 107] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=220741930162, lastReceivedTime=220708170330, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-12 20:55:55,209] - TCPPacketReceiver::processPendingEvents(192.168.56.1:58815): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 20:55:55,212] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=220742168288, lastReceivedTime=220708393397, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-12 20:55:55,225] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-12 20:55:55,226] - onLoginAppShutDown()
  INFO root [2019-09-12 20:55:55,226] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-12 20:55:55,226] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-12 20:55:55,240] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-12 20:55:55,241] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-12 20:55:55,542] - ThreadPool::destroy(): waiting for thread(2)[0x87e950,0x90b9e0,], try=1
  INFO root [2019-09-12 20:58:52,278] - 
  INFO root [2019-09-12 20:58:52,287] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 20:58:52,288] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 20:58:52,289] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 20:58:52,289] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 20:58:52,305] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 20:58:52,305] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 20:58:52,322] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:58853 )
  INFO root [2019-09-12 20:58:52,322] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:58853, SOMAXCONN=128.
  INFO root [2019-09-12 20:58:52,322] - server-configs:
  INFO root [2019-09-12 20:58:52,322] - 	internalTcpAddr : 192.168.56.1:58853
  INFO root [2019-09-12 20:58:52,322] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 20:58:52,323] - 	componentID : 9000
  WARN root [2019-09-12 20:58:52,558] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-12 20:58:54,064] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 14624 ----
  INFO root [2019-09-12 20:58:54,071] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 20:58:54,072] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 20:58:54,072] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 20:58:54,284] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 20:58:54,285] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 20:58:54,298] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 20:58:54,299] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 20:58:54,728] - ---- loginapp is running ----
 DEBUG root [2019-09-12 20:58:54,729] - Components::process(): Request for the process of identity...
  INFO root [2019-09-12 20:59:01,959] - Components::findComponents: find logger(1)...
  INFO root [2019-09-12 20:59:02,031] - Components::findComponents: found logger, addr:192.168.56.1:58852
  INFO root [2019-09-12 20:59:02,032] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:59:02,032] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:59:02,033] - Components::findComponents: find dbmgr(2)...
  INFO root [2019-09-12 20:59:02,137] - Components::findComponents: found dbmgr, addr:192.168.56.1:58858
  INFO root [2019-09-12 20:59:02,137] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 20:59:02,138] - Components::addComponent[dbmgr], uid=2082, componentID=4000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 20:59:02,138] - Components::findComponents: found dbmgr, addr:192.168.56.1:58858
  WARN root [2019-09-12 20:59:03,139] - Forwarding logs to logger[192.168.56.1:58852/0/0/0]...
 ERROR root [2019-09-12 21:01:34,086] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=221504635693, lastReceivedTime=221477995687, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-12 21:01:34,092] - The following logs sent to logger failed:
  WARN root [2019-09-12 21:01:34,092] - ==> [2019-09-12 21:01:34 085] TCPPacketReceiver::processPendingEvents(192.168.56.1:58852): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 21:01:34,093] - ==> [2019-09-12 21:01:34 087] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=221504635693, lastReceivedTime=221477995687, inactivityExceptionPeriod=134887628)
  WARN root [2019-09-12 21:01:34,187] - TCPPacketReceiver::processPendingEvents(192.168.56.1:58858): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 21:01:34,190] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=221504868896, lastReceivedTime=221478385634, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-12 21:01:34,194] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-12 21:01:34,195] - onLoginAppShutDown()
  INFO root [2019-09-12 21:01:34,195] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-12 21:01:34,195] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-12 21:01:34,212] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-12 21:01:34,212] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-12 21:01:34,513] - ThreadPool::destroy(): waiting for thread(2)[0x23d0e0,0x23cc00,], try=1
  INFO root [2019-09-12 21:01:58,045] - 
  INFO root [2019-09-12 21:01:58,057] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 21:01:58,058] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 21:01:58,058] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 21:01:58,058] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 21:01:58,107] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 21:01:58,107] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 21:01:58,115] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:58897 )
  INFO root [2019-09-12 21:01:58,115] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:58897, SOMAXCONN=128.
  INFO root [2019-09-12 21:01:58,115] - server-configs:
  INFO root [2019-09-12 21:01:58,116] - 	internalTcpAddr : 192.168.56.1:58897
  INFO root [2019-09-12 21:01:58,116] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 21:01:58,116] - 	componentID : 9000
  WARN root [2019-09-12 21:01:58,328] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-12 21:01:58,600] - Components::findLogger: found logger, addr:192.168.56.1:58893
  INFO root [2019-09-12 21:01:58,600] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 21:01:58,600] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 21:01:58,605] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 8756 ----
  INFO root [2019-09-12 21:01:58,606] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 21:01:58,606] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 21:01:58,606] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 21:01:58,808] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 21:01:58,808] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 21:01:58,821] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 21:01:58,821] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 21:01:59,218] - ---- loginapp is running ----
 DEBUG root [2019-09-12 21:01:59,219] - Components::process(): Request for the process of identity...
  WARN root [2019-09-12 21:02:05,346] - Forwarding logs to logger[192.168.56.1:58893/0/0/0]...
 ERROR root [2019-09-12 21:03:35,746] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=221778371051, lastReceivedTime=221761144188, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-12 21:03:35,753] - The following logs sent to logger failed:
  WARN root [2019-09-12 21:03:35,753] - ==> [2019-09-12 21:03:35 745] TCPPacketReceiver::processPendingEvents(192.168.56.1:58893): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 21:03:35,754] - ==> [2019-09-12 21:03:35 747] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=221778371051, lastReceivedTime=221761144188, inactivityExceptionPeriod=134887437)
  WARN root [2019-09-12 21:03:35,920] - TCPPacketReceiver::processPendingEvents(192.168.56.1:58894): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 21:03:35,927] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=221778776813, lastReceivedTime=221761154372, inactivityExceptionPeriod=134887437)
  INFO root [2019-09-12 21:03:35,957] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-12 21:03:35,962] - onLoginAppShutDown()
  INFO root [2019-09-12 21:03:35,962] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-12 21:03:35,962] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-12 21:03:35,984] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-12 21:03:35,984] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-12 21:03:36,285] - ThreadPool::destroy(): waiting for thread(2)[0x2e5228,0x2e5430,], try=1
 DEBUG root [2019-09-12 21:03:36,691] - TPThread::~TPThread(): 0x2e5228
 DEBUG root [2019-09-12 21:03:36,692] - TPThread::~TPThread(): 0x2e5430
 DEBUG root [2019-09-12 21:03:36,692] - ThreadPool::destroy(): successfully!
 DEBUG root [2019-09-12 21:03:36,720] - Bundle::destroyObjPool(): size 20.
 DEBUG root [2019-09-12 21:03:36,720] - Channel::destroyObjPool(): size 15.
 DEBUG root [2019-09-12 21:03:36,721] - TCPPacket::destroyObjPool(): size 3.
 DEBUG root [2019-09-12 21:03:36,722] - UDPPacket::destroyObjPool(): size 2.
 DEBUG root [2019-09-12 21:03:36,722] - EndPoint::destroyObjPool(): size 14.
 DEBUG root [2019-09-12 21:03:36,723] - Address::destroyObjPool(): size 0.
 DEBUG root [2019-09-12 21:03:36,723] - TCPPacketReceiver::destroyObjPool(): size 0.
 DEBUG root [2019-09-12 21:03:36,723] - UDPPacketReceiver::destroyObjPool(): size 0.
  INFO root [2019-09-12 21:03:36,723] - loginapp(9000) has shut down.
  INFO root [2019-09-12 21:05:04,222] - 
  INFO root [2019-09-12 21:05:04,226] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 21:05:04,235] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 21:05:04,235] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 21:05:04,235] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 21:05:04,307] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 21:05:04,307] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 21:05:04,321] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:58938 )
  INFO root [2019-09-12 21:05:04,322] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:58938, SOMAXCONN=128.
  INFO root [2019-09-12 21:05:04,322] - server-configs:
  INFO root [2019-09-12 21:05:04,322] - 	internalTcpAddr : 192.168.56.1:58938
  INFO root [2019-09-12 21:05:04,322] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 21:05:04,322] - 	componentID : 9000
  WARN root [2019-09-12 21:05:04,653] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-12 21:05:06,163] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 7492 ----
  INFO root [2019-09-12 21:05:06,167] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 21:05:06,170] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 21:05:06,170] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 21:05:06,542] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 21:05:06,542] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 21:05:06,557] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 21:05:06,558] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 21:05:07,196] - ---- loginapp is running ----
 DEBUG root [2019-09-12 21:05:07,197] - Components::process(): Request for the process of identity...
  INFO root [2019-09-12 21:05:14,518] - Components::findComponents: find logger(1)...
  INFO root [2019-09-12 21:05:14,585] - Components::findComponents: found logger, addr:192.168.56.1:58943
  INFO root [2019-09-12 21:05:14,585] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 21:05:14,585] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 21:05:14,587] - Components::findComponents: find dbmgr(2)...
  INFO root [2019-09-12 21:05:14,593] - Components::findComponents: found logger, addr:192.168.56.1:58943
  INFO root [2019-09-12 21:05:14,596] - Components::findComponents: found dbmgr, addr:192.168.56.1:58940
  INFO root [2019-09-12 21:05:14,596] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 21:05:14,596] - Components::addComponent[dbmgr], uid=2082, componentID=4000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 21:05:14,697] - Components::findComponents: found dbmgr, addr:192.168.56.1:58940
  WARN root [2019-09-12 21:05:15,698] - Forwarding logs to logger[192.168.56.1:58943/0/0/0]...
 ERROR root [2019-09-12 21:07:02,511] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=222243592796, lastReceivedTime=222180431206, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-12 21:07:02,518] - The following logs sent to logger failed:
  WARN root [2019-09-12 21:07:02,518] - ==> [2019-09-12 21:07:02 510] TCPPacketReceiver::processPendingEvents(192.168.56.1:58943): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 21:07:02,518] - ==> [2019-09-12 21:07:02 512] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=222243592796, lastReceivedTime=222180431206, inactivityExceptionPeriod=134887564)
  WARN root [2019-09-12 21:07:02,614] - TCPPacketReceiver::processPendingEvents(192.168.56.1:58940): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 21:07:02,621] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=222243841080, lastReceivedTime=222180267393, inactivityExceptionPeriod=134887564)
  INFO root [2019-09-12 21:07:02,627] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-12 21:07:02,628] - onLoginAppShutDown()
  INFO root [2019-09-12 21:07:02,629] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-12 21:07:02,629] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-12 21:07:02,640] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-12 21:07:02,641] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-12 21:07:02,942] - ThreadPool::destroy(): waiting for thread(2)[0x1a3d308,0x1a3d370,], try=1
  INFO root [2019-09-12 21:11:18,925] - 
  INFO root [2019-09-12 21:11:18,928] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 21:11:18,929] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 21:11:18,929] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 21:11:18,929] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 21:11:18,985] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 21:11:18,985] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 21:11:18,989] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:58988 )
  INFO root [2019-09-12 21:11:18,989] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:58988, SOMAXCONN=128.
  INFO root [2019-09-12 21:11:19,012] - server-configs:
  INFO root [2019-09-12 21:11:19,013] - 	internalTcpAddr : 192.168.56.1:58988
  INFO root [2019-09-12 21:11:19,013] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 21:11:19,013] - 	componentID : 9000
  WARN root [2019-09-12 21:11:19,270] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-12 21:11:19,352] - Components::findLogger: found logger, addr:192.168.56.1:58984
  INFO root [2019-09-12 21:11:19,352] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 21:11:19,353] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 21:11:19,358] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 10808 ----
  INFO root [2019-09-12 21:11:19,358] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 21:11:19,359] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 21:11:19,359] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 21:11:19,592] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 21:11:19,592] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 21:11:19,604] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 21:11:19,605] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 21:11:20,028] - ---- loginapp is running ----
 DEBUG root [2019-09-12 21:11:20,029] - Components::process(): Request for the process of identity...
  WARN root [2019-09-12 21:11:26,153] - Forwarding logs to logger[192.168.56.1:58984/0/0/0]...
 ERROR root [2019-09-12 21:13:23,701] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=223101268927, lastReceivedTime=223090232244, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-12 21:13:23,704] - The following logs sent to logger failed:
  WARN root [2019-09-12 21:13:23,704] - ==> [2019-09-12 21:13:23 697] TCPPacketReceiver::processPendingEvents(192.168.56.1:58984): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 21:13:23,705] - ==> [2019-09-12 21:13:23 702] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=223101268927, lastReceivedTime=223090232244, inactivityExceptionPeriod=134887628)
  WARN root [2019-09-12 21:13:23,810] - TCPPacketReceiver::processPendingEvents(192.168.56.1:58981): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 21:13:23,813] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=223101520527, lastReceivedTime=223090110888, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-12 21:13:23,829] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-12 21:13:23,841] - onLoginAppShutDown()
  INFO root [2019-09-12 21:13:23,842] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-12 21:13:23,842] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-12 21:13:23,856] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-12 21:13:23,856] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-12 21:13:24,157] - ThreadPool::destroy(): waiting for thread(2)[0x19b5370,0x19b5780,], try=1
  INFO root [2019-09-12 21:15:41,070] - 
  INFO root [2019-09-12 21:15:41,072] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 21:15:41,073] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 21:15:41,074] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 21:15:41,074] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 21:15:41,109] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 21:15:41,109] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 21:15:41,113] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:59026 )
  INFO root [2019-09-12 21:15:41,113] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:59026, SOMAXCONN=128.
  INFO root [2019-09-12 21:15:41,114] - server-configs:
  INFO root [2019-09-12 21:15:41,114] - 	internalTcpAddr : 192.168.56.1:59026
  INFO root [2019-09-12 21:15:41,114] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 21:15:41,114] - 	componentID : 9000
  WARN root [2019-09-12 21:15:41,384] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-12 21:15:41,689] - Components::findLogger: found logger, addr:192.168.56.1:59025
  INFO root [2019-09-12 21:15:41,689] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 21:15:41,690] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 21:15:41,695] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 2816 ----
  INFO root [2019-09-12 21:15:41,695] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 21:15:41,695] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 21:15:41,696] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 21:15:41,891] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 21:15:41,891] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 21:15:41,903] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 21:15:41,903] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 21:15:42,294] - ---- loginapp is running ----
 DEBUG root [2019-09-12 21:15:42,295] - Components::process(): Request for the process of identity...
  WARN root [2019-09-12 21:15:48,420] - Forwarding logs to logger[192.168.56.1:59025/0/0/0]...
 ERROR root [2019-09-12 21:17:02,640] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=223593884433, lastReceivedTime=223545426866, inactivityExceptionPeriod=134887197)
  INFO root [2019-09-12 21:17:02,643] - The following logs sent to logger failed:
  WARN root [2019-09-12 21:17:02,643] - ==> [2019-09-12 21:17:02 639] TCPPacketReceiver::processPendingEvents(192.168.56.1:59025): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 21:17:02,643] - ==> [2019-09-12 21:17:02 641] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=223593884433, lastReceivedTime=223545426866, inactivityExceptionPeriod=134887197)
  WARN root [2019-09-12 21:17:02,763] - TCPPacketReceiver::processPendingEvents(192.168.56.1:59023): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 21:17:02,767] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=223594168532, lastReceivedTime=223545335990, inactivityExceptionPeriod=134887197)
  INFO root [2019-09-12 21:17:02,775] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-12 21:17:02,777] - onLoginAppShutDown()
  INFO root [2019-09-12 21:17:02,777] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-12 21:17:02,777] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-12 21:17:02,793] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-12 21:17:02,793] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-12 21:17:03,094] - ThreadPool::destroy(): waiting for thread(2)[0x1ef0a78,0x1ef0e88,], try=1
  INFO root [2019-09-12 21:19:24,026] - 
  INFO root [2019-09-12 21:19:24,031] - -----------------------------------------------------------------------------------------


  INFO root [2019-09-12 21:19:24,032] - Resmgr::initialize: KBE_ROOT=D:/kbengine/
  INFO root [2019-09-12 21:19:24,032] - Resmgr::initialize: KBE_RES_PATH=D:/kbengine/kbe/res/;D:/kbengine/BattleOfBalls_assets/;D:/kbengine/BattleOfBalls_assets/scripts/;D:/kbengine/BattleOfBalls_assets/res/
  INFO root [2019-09-12 21:19:24,032] - Resmgr::initialize: KBE_BIN_PATH=D:/kbengine/kbe/bin/server/
  INFO root [2019-09-12 21:19:24,061] - NetworkInterface::initialize(EXTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:20013 )
  INFO root [2019-09-12 21:19:24,061] - NetworkInterface::initialize(EXTERNAL-TCP): address 192.168.56.1:20013, SOMAXCONN=128.
  INFO root [2019-09-12 21:19:24,065] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.56.1 ( 192.168.56.1:59065 )
  INFO root [2019-09-12 21:19:24,065] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.56.1:59065, SOMAXCONN=128.
  INFO root [2019-09-12 21:19:24,065] - server-configs:
  INFO root [2019-09-12 21:19:24,065] - 	internalTcpAddr : 192.168.56.1:59065
  INFO root [2019-09-12 21:19:24,066] - 	externalTcpAddr : 192.168.56.1:20013
  INFO root [2019-09-12 21:19:24,066] - 	componentID : 9000
  WARN root [2019-09-12 21:19:24,289] - invalid UID(0) <= 0, please check UID for environment! automatically set to 2082.
  INFO root [2019-09-12 21:19:24,509] - Components::findLogger: found logger, addr:192.168.56.1:59058
  INFO root [2019-09-12 21:19:24,510] - Components::addComponent: reset orderLog, uid=2082!
  INFO root [2019-09-12 21:19:24,510] - Components::addComponent[logger], uid=2082, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [2019-09-12 21:19:24,516] - ---- loginapp Version: 2.4.5. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: CE9FF53F4E85387EE4455C7191269324. Config: Debug X86. Built: 09:58:16 May  8 2019. AppID: 9000. UID: 2082. PID: 13960 ----
  INFO root [2019-09-12 21:19:24,516] - Server error_name: Intel(R) Core(TM) i5-6300HQ CPU @ 2.30GHz with 2097151kB RAM
  INFO root [2019-09-12 21:19:24,517] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [2019-09-12 21:19:24,517] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [2019-09-12 21:19:24,712] - Script::install(): is successfully, Python=(3.7.3 (default, May  8 2019, 09:53:38) [MSC v.1916 32 bit (Intel)])!
  INFO root [2019-09-12 21:19:24,712] - Script::install(): KBExtra is successfully!
S_INFO root [2019-09-12 21:19:24,725] - onLoginAppReady: bootstrapGroupIndex=0, bootstrapGlobalIndex=0
  INFO root [2019-09-12 21:19:24,725] - TelnetServer server is running on port 31000
  INFO root [2019-09-12 21:19:25,130] - ---- loginapp is running ----
 DEBUG root [2019-09-12 21:19:25,130] - Components::process(): Request for the process of identity...
  WARN root [2019-09-12 21:19:31,280] - Forwarding logs to logger[192.168.56.1:59058/0/0/0]...
 ERROR root [2019-09-12 21:20:44,715] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=224093553799, lastReceivedTime=224046582140, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-12 21:20:44,720] - The following logs sent to logger failed:
  WARN root [2019-09-12 21:20:44,721] - ==> [2019-09-12 21:20:44 713] TCPPacketReceiver::processPendingEvents(192.168.56.1:59058): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 21:20:44,721] - ==> [2019-09-12 21:20:44 717] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=224093553799, lastReceivedTime=224046582140, inactivityExceptionPeriod=134887628)
  WARN root [2019-09-12 21:20:44,892] - TCPPacketReceiver::processPendingEvents(192.168.56.1:59061): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [2019-09-12 21:20:44,897] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=224093961462, lastReceivedTime=224067766020, inactivityExceptionPeriod=134887628)
  INFO root [2019-09-12 21:20:44,912] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
S_INFO root [2019-09-12 21:20:44,926] - onLoginAppShutDown()
  INFO root [2019-09-12 21:20:44,926] - Shutdowner::onShutdown: shutting down
  INFO root [2019-09-12 21:20:44,927] - Shutdowner::onShutdownEnd: shutting down
  INFO root [2019-09-12 21:20:44,945] - Script::uninstall(): is successfully!
 DEBUG root [2019-09-12 21:20:44,946] - ThreadPool::destroy(): starting size 2.
  WARN root [2019-09-12 21:20:45,247] - ThreadPool::destroy(): waiting for thread(2)[0x1b60d98,0x1b60ac0,], try=1
 DEBUG root [2019-09-12 21:20:45,653] - TPThread::~TPThread(): 0x1b60d98
 DEBUG root [2019-09-12 21:20:45,654] - TPThread::~TPThread(): 0x1b60ac0
 DEBUG root [2019-09-12 21:20:45,654] - ThreadPool::destroy(): successfully!
 DEBUG root [2019-09-12 21:20:45,680] - Bundle::destroyObjPool(): size 20.
 DEBUG root [2019-09-12 21:20:45,681] - Channel::destroyObjPool(): size 14.
 DEBUG root [2019-09-12 21:20:45,682] - TCPPacket::destroyObjPool(): size 3.
 DEBUG root [2019-09-12 21:20:45,682] - UDPPacket::destroyObjPool(): size 2.
 DEBUG root [2019-09-12 21:20:45,682] - EndPoint::destroyObjPool(): size 13.
 DEBUG root [2019-09-12 21:20:45,683] - Address::destroyObjPool(): size 0.
 DEBUG root [2019-09-12 21:20:45,686] - TCPPacketReceiver::destroyObjPool(): size 0.
 DEBUG root [2019-09-12 21:20:45,686] - UDPPacketReceiver::destroyObjPool(): size 0.
  INFO root [2019-09-12 21:20:45,686] - loginapp(9000) has shut down.
